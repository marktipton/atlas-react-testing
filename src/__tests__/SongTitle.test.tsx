import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import  SongTitle  from "../components/SongTitle";

test("volume control renders correctly w/", () => {
  const { container } = render(<SongTitle />);
  expect(container).toMatchSnapshot();
});

test("renders songtitle component w/ given title and artist", () => {
  const { container } = render(<SongTitle title="Sandstorm" artist="Darude"/>);
  expect(container).toMatchSnapshot();
  expect(screen.getByText("Sandstorm")).toBeTruthy();
  expect(screen.getByText("Darude")).toBeTruthy();
});

test("renders SongTitle component with empty artist and title", () => {
  const { container } = render(<SongTitle title="" artist="" />);
  expect(container).toMatchSnapshot();

  // Check that the <h1> element is rendered but contains no text
  const titleElement = container.querySelector('h1');
  expect(titleElement).not.toBeNull();
  expect(titleElement?.textContent).toBe("");

  // Check that the <p> element is rendered but contains no text
  const artistElement = container.querySelector('p');
  expect(artistElement).not.toBeNull();
  expect(artistElement?.textContent).toBe("");
});
