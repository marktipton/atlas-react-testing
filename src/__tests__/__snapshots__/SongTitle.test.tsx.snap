// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders SongTitle component with empty artist and title 1`] = `
<div>
  <div>
    <h1
      class="font-bold text-xl text-primary"
    />
    <p
      class="text-secondary"
    />
  </div>
</div>
`;

exports[`renders songtitle component w/ given title and artist 1`] = `
<div>
  <div>
    <h1
      class="font-bold text-xl text-primary"
    >
      Sandstorm
    </h1>
    <p
      class="text-secondary"
    >
      Darude
    </p>
  </div>
</div>
`;
